src/interval.jl

```diff
@@ -482,6 +714,103 @@ function Base.merge(a::AbstractInterval, b::AbstractInterval)
     return Interval(left, right)
 end
 
+Base.union(x::AbstractInterval) = x
+
+const AbstractIntervals = Union{AbstractInterval, AbstractVector{<:AbstractInterval}}
+mergeclean(x, y) = unbunch(union(x), union(y))
+function Base.intersect(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && iny, mergeclean(x, y)...)
+end
+function Base.union(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx || iny, mergeclean(x, y)...)
+end
+function Base.setdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && !iny, mergeclean(x, y)...)
+end
+function Base.symdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx ⊻ iny, mergeclean(x, y)...)
+end
+function Base.issubset(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(setdiff(x, y))
+end
+# may or may not be from Base (see top of `Intervals.jl`)
+function isdisjoint(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(intersect(x, y))
+end
+Base.in(x::AbstractInterval, y::AbstractVector{<:AbstractInterval}) = any(yᵢ -> x ∈ yᵢ, y)
+function Base.issetequal(x::AbstractIntervals, y::AbstractIntervals)
+    x, y = mergeclean(x,y)
+    return x == y
+end
+
+asarray(x) = [x]
+asarray(x::AbstractArray) = x
+alength(x) = 1
+alength(x::AbstractArray) = length(x)
```

Comment:
I'm guessing you want `length` from an `Interval`? As you're only using this once probably makes more sense to use multiple dispatch on `intersectmap`. Alternatively, defining `length(::AbstractInterval)` would be a better solution but also seems unnecessary